kubernetes/skaffold/src/main/kotlin/com/google/kubernetes/tools/skaffold/SkaffoldExecutorService.kt

```diff
@@ -177,10 +184,21 @@ data class SkaffoldExecutorSettings(
 data class SkaffoldProcess(val process: Process, val commandLine: String)
 
 /**
- * Default implementation of Skaffold execution service, where Skaffold executable is assumed to
- * be already installed on the system and be available in PATH.
+ * Default implementation of Skaffold executor service. The Skaffold executable is first pulled
+ * from the Kubernetes settings, and then the PATH if unavailable in the settings.
  */
 class DefaultSkaffoldExecutorService : SkaffoldExecutorService() {
-    // use executable available in PATH
-    override var skaffoldExecutablePath: Path = Paths.get("skaffold")
+    /**
+     * First try to find Skaffold in the Kubernetes settings under Google -> Kubernetes.
+     * If not found, then try to find it on the user's PATH.
+     */
+    override fun getSkaffoldExecutablePath(): Path {
```

Comment:
Can we keep the property but provide a custom getter?

Reply:
Yep. That works

Reply:
Yep. That works

Reply:
hmm with a getter, I'm [forced to initialize it](https://stackoverflow.com/questions/41440750/kotlin-why-do-i-need-to-initialize-a-var-with-custom-getter), it seems. Something like this which isn't very nice:

```
override var skaffoldExecutablePath: Path? = null
  get() {
    // ...
  }
```

Reply:
we can probably just initialize it to `Paths.get(skaffold)` but I'm ok with either way

Reply:
thats true. I'll do that.

Reply:
done - refactored to custom getter using `field` accessor